nifi-nar-bundles/nifi-jslt-bundle/nifi-jslt-processors/src/main/java/org/apache/nifi/processors/jslt/JSLTTransformJSON.java

```diff
@@ -107,6 +108,16 @@ public class JSLTTransformJSON extends AbstractProcessor {
             .defaultValue(EACH_OBJECT.getValue())
             .build();
 
+    public static final PropertyDescriptor RESULT_FILTER = new PropertyDescriptor.Builder()
+            .name("jslt-transform-result-filter")
+            .displayName("Transform Result Filter")
+            .description("An optional filter of output results using another JSLT, allowing you to change the default filter."
+                    + " The default filter is \". != null and . != {} and . != []\" which excludes objects with null values, empty objects and empty arrays")
```

Comment:
Instead of making this an optional property, what do you think about setting it as required and providing the default value? This would make the default behavior more obvious, and provide a quick example of what it can do.

Reply:
I like your suggested approach, and I will make the change.  I certainly didn't know that the JSLT library had a default output filter, and it would be nice to make this obvious.  I'm a bit concerned about performance of using a JSLT filter instead of the default filter, so I'll investigate whether that makes a significant difference.

Reply:
@exceptionfactory I made the new property required, gave it a default value, and updated the description.  My performance tests showed a small penalty (roughly 5%) using a simple JSLT filter over the built-in default.  To mitigate this, if the property is set to the default filter value, then I use the library's built-in default filter.  This should preserve current performance.

Reply:
Thanks for the updates @mosermw. Can you clarify the difference between the "default filter" and the default configured value of the new property? I may have misunderstood the initial property description. Is the difference based on the way the internal "default filter" is implemented, versus the same functional result using the property default JSLT expression?

Reply:
No problem @exceptionfactory it took me a bit of reading and testing to learn this.  The default filter in the JSLT library is  [DefaultJsonFilter.java](https://github.com/schibsted/jslt/blob/master/core/src/main/java/com/schibsted/spt/data/jslt/filters/DefaultJsonFilter.java).  It uses the [NodeUtils::isValue](https://github.com/schibsted/jslt/blob/afb0cc037daac1ac69d9348d42a62d09a12cc4f5/core/src/main/java/com/schibsted/spt/data/jslt/impl/NodeUtils.java#L59) method to determine whether to output a JsonNode.  This method is a bit faster than processing a JSLT that does the same thing.

The [documentation about this filter](https://github.com/schibsted/jslt/blob/master/docs/api.md#object-key-filter) shows an equivalent JSLT which is what you see in the JSLT_FILTER_DEFAULT variable.  Even though implementing a Java based filter would be faster, I think using JSLT for the filter gives users maximum flexibility.

Reply:
Thanks for the clarification and references @mosermw, that makes sense now. I agree that using JSLT for the filter definition itself makes the most sense from a configuration point of view.